pinot-common/src/main/java/org/apache/pinot/common/datatable/DataTable.java

```diff
@@ -134,11 +134,12 @@ enum MetadataKey {
     OPERATOR_EXECUTION_TIME_MS(30, "operatorExecutionTimeMs", MetadataValueType.LONG),
     OPERATOR_ID(31, "operatorId", MetadataValueType.STRING),
     OPERATOR_EXEC_START_TIME_MS(32, "operatorExecStartTimeMs", MetadataValueType.LONG),
-    OPERATOR_EXEC_END_TIME_MS(33, "operatorExecEndTimeMs", MetadataValueType.LONG);
+    OPERATOR_EXEC_END_TIME_MS(33, "operatorExecEndTimeMs", MetadataValueType.LONG),
+    NUM_JOIN_LIMIT_REACHED(34, "numJoinsLimitReached", MetadataValueType.STRING);
```

Comment:
@saurabhd336 this might affect the partial result flag setting.

Reply:
To be more specific. `numJoinsLimit` can cause confusion. Modify other places accordingly
```suggestion
    MAX_ROWS_IN_JOIN_REACHED(34, "maxRowsInJoinReached", MetadataValueType.STRING);
```

Reply:
good idea. should we be even more specific: `MAX_ROWS_IN_JOIN_HASH_TABLE_REACHED` ?

Reply:
Sounds good to me. Maybe `MAX_ROWS_IN_JOIN_TABLE_REACHED` since we might optimize it to not use hash table?

Reply:
wilco